component/status.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component // import "go.opentelemetry.io/collector/component"
+
+import "go.opentelemetry.io/collector/config"
+
+type StatusReport struct {
+	ComponentID config.ComponentID
+	Error       error
+}
+
+type StatusReportFunc func(status StatusReport)
+
+type StatusReporters struct {
```

Comment:
I'm not a big fan of having this as a public API. Perhaps it should be moved to the service package and kept local to it?